We are providing the full path to the key via keydir, while mkimage expects base attributes and sets object manually to keyname.
keyname cannot be used in our case as it is fixed to "db" due to old compatibility and we cannot really adjust it in build process
diff -aurN a/lib/rsa/rsa-sign.c b/lib/rsa/rsa-sign.c
--- a/lib/rsa/rsa-sign.c	2021-09-10 09:26:04.667575205 +0200
+++ b/lib/rsa/rsa-sign.c	2021-09-10 09:31:54.998347877 +0200
@@ -135,8 +135,8 @@
 	if (engine_id && !strcmp(engine_id, "pkcs11")) {
 		if (keydir)
 			snprintf(key_id, sizeof(key_id),
-				 "pkcs11:%s;object=%s;type=public",
-				 keydir, name);
+				 "pkcs11:%s;type=public",
+				 keydir);
 		else
 			snprintf(key_id, sizeof(key_id),
 				 "pkcs11:object=%s;type=public",
@@ -255,8 +255,8 @@
 	if (engine_id && !strcmp(engine_id, "pkcs11")) {
 		if (keydir)
 			snprintf(key_id, sizeof(key_id),
-				 "pkcs11:%s;object=%s;type=private",
-				 keydir, name);
+				 "pkcs11:%s;type=private",
+				 keydir);
 		else
 			snprintf(key_id, sizeof(key_id),
 				 "pkcs11:object=%s;type=private",
